theano/sandbox/cuda/dnn.py

```diff
@@ -1054,6 +1079,24 @@ def c_code(self, node, name, inputs, outputs, sub):
   PyErr_Format(PyExc_RuntimeError,
                "GpuDnnPoolGrad: error doing operation: %%s",
                cudnnGetErrorString(err%(name)s));
+  printf("input.shape=(%%d, %%d, %%d, %%d) input_grad.shape=(%%d, %%d, %%d, %%d) output.shape=(%%d, %%d, %%d, %%d) output_grad.shape=(%%d, %%d, %%d, %%d)\\n",
+CudaNdarray_HOST_DIMS(%(input)s)[0],
+CudaNdarray_HOST_DIMS(%(input)s)[1],
+CudaNdarray_HOST_DIMS(%(input)s)[2],
+CudaNdarray_HOST_DIMS(%(input)s)[3],
+CudaNdarray_HOST_DIMS(%(input_grad)s)[0],
+CudaNdarray_HOST_DIMS(%(input_grad)s)[1],
+CudaNdarray_HOST_DIMS(%(input_grad)s)[2],
+CudaNdarray_HOST_DIMS(%(input_grad)s)[3],
+CudaNdarray_HOST_DIMS(%(output)s)[0],
+CudaNdarray_HOST_DIMS(%(output)s)[1],
+CudaNdarray_HOST_DIMS(%(output)s)[2],
+CudaNdarray_HOST_DIMS(%(output)s)[3],
+CudaNdarray_HOST_DIMS(%(output_grad)s)[0],
+CudaNdarray_HOST_DIMS(%(output_grad)s)[1],
+CudaNdarray_HOST_DIMS(%(output_grad)s)[2],
+CudaNdarray_HOST_DIMS(%(output_grad)s)[3]
+        );
```

Comment:
I still don't like printfs, but at least it's in an error path.  Making this part of the error message would be better.